src/components/MessagesList/MessagesList.vue

```diff
@@ -329,40 +347,85 @@ export default {
 
 	methods: {
 		prepareMessagesGroups(messages) {
-			const groups = []
+			let prevGroupMap = {}
+			const groupsByDate = {}
 			let lastMessage = null
+			let dateSeparator = null
+			let groupId = null
 			for (const message of messages) {
 				if (!this.messagesShouldBeGrouped(message, lastMessage)) {
-					if (groups.at(-1)) {
-						groups.at(-1).nextMessageId = message.id
+					// If the message should be grouped, we create a new group
+					dateSeparator = this.generateDateSeparator(message)
+					groupId = message.id
+
+					if (!groupsByDate[dateSeparator]) {
+						groupsByDate[dateSeparator] = {}
```

Comment:
```suggestion
					const dateTimestamp = moment(message.timestamp * 1000).startOf('day').unix()
					if (!groupsByDate[dateTimestamp]) {
						groupsByDate[dateTimestamp] = {}
```

Reply:
Don't think it's necessary. We have usually 1-3 separators, depending on conversation activity. Also timesstamps are numeric, and if you iterate object wtih timestamp keys, it should be already sorted correctly

Reply:
Ok and then just generate the separator in the div